### PR TITLE
add default selection support

### PR DESCRIPTION
**Pull request by jamestalton**
Support for a view to have default selection

This is not hoked up to any dialogs yet, this is in preparation for future work.

In the case where you are opening a dialog from say the Teams view,
you create an AWXView for the dialogs table with the currently selected items using useAwxView({ defaultSelections: view.selected })

Or from a PageFormSelect where the current value in the select is passed to a SelectDialog by creating a new AwxView with the current item as the defaultSelections

### CODE REVIEW

File: framework/PageTable/useTableItems.tsx [hunk 1]
```diff
@@ -92,9 +92,19 @@ export interface ISelected<T extends object> {
 
 export function useSelected<T extends object>(
   items: T[],
-  keyFn: (item: T) => string | number
+  keyFn: (item: T) => string | number,
+  defaultSelection?: T[]
```
Comment (AlexSCorey) [inline framework/PageTable/useTableItems.tsx:96]: Will this function every time the modal is opened?  If so, I  think  `defaultSelection` is a misnomer and should be changed to something like `selectedItems`, or `value`.

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
@jamestalton I checked out this branch locally, and it there is still a disconnect between the form field value, and the selected items in the modal.  See gif.
selectedValues

File: framework/PageTable/useTableItems.tsx (hunk 1 above)
Comment (jamestalton) [reply framework/PageTable/useTableItems.tsx:96]: This PR does not hook up the default selection for the view into any dialogs.
This just adds the support for the view and future PRs can take advantage of that.

**Review by vidyanambiar — COMMENTED** [review]
Looks good 👍  

It would help to have some documentation added around this and perhaps an example scenario.

**Review by vidyanambiar — APPROVED** [review]

**Review by AlexSCorey — APPROVED** [review]

**Pull request merged**